dts/bindings/base/base.yaml

```diff
@@ -3,6 +3,10 @@
 include: [pm.yaml]
 
 properties:
+  zephyr,device-type:
+    type: string
+    description: indicates the device type.
```

Comment:
I'm sorry, but to me, using devicetree to add yet more software properties is a -1. Usage of `zephyr,` has become a mess to a point where we should likely rename `devicetree` to something else, maybe `ztree` or something. This solution also doesn't account for device nodes that can have >1 device type, which even if not supported now, they should if we ever pretend to fully align with Linux.

Reply:
adding that my suggestion just above would theoretically be able to handle multiple types